### ISSUE TITLE
Bug: logger breaks on nodejs Docker lambdas since 2.4.0

### ISSUE THREAD

**Issue by ralbertazzi** [aws-powertools/powertools-lambda-typescript#2851]
The base image used to build Docker images, `public.ecr.aws/lambda/nodejs:20`, sets the environment variable `TZ=:/etc/localtime`. This value is not recognised as valid one and throws an error. I believe this was introduced by [this PR](https://github.com/aws-powertools/powertools-lambda-typescript/pull/2710) that was merged in 2.4.0.

### Expected Behavior

Building a Docker Lambda using the logger library without further configuration, i.e. I'd expect the library to print times in UTC when TZ is `:/etc/localtime`

### Current Behavior

`RangeError: Invalid time zone specified: :/etc/localtime` is raised when logging anything under this condition

### Code snippet

```typescript
import { Logger } from "@aws-lambda-powertools/logger";

const log = new Logger();
log.info("Hello");
```

### Steps to Reproduce

1. Build a Docker Lambda from `public.ecr.aws/lambda/nodejs:20`
2. Install the logger library
3. Log something

### Possible Solution

The library should print times in UTC when TZ is `:/etc/localtime`

### Powertools for AWS Lambda (TypeScript) version

latest

### AWS Lambda function runtime

20.x

### Packaging format used

npm

### Execution logs

```Shell
RangeError: Invalid time zone specified: :/etc/localtime
    at new DateTimeFormat (<anonymous>)
    at #getDateFormatter (/var/task/node_modules/.pnpm/@aws-lambda-powertools+logger@2.5.0/node_modules/@aws-lambda-powertools/logger/lib/cjs/formatter/LogFormatter.js:85:16)
    at #generateISOTimestampWithOffset (/var/task/node_modules/.pnpm/@aws-lambda-powertools+logger@2.5.0/node_modules/@aws-lambda-powertools/logger/lib/cjs/formatter/LogFormatter.js:103:82)
    at PowertoolsLogFormatter.formatTimestamp (/var/task/node_modules/.pnpm/@aws-lambda-powertools+logger@2.5.0/node_modules/@aws-lambda-powertools/logger/lib/cjs/formatter/LogFormatter.js:53:56)
    at PowertoolsLogFormatter.formatAttributes (/var/task/node_modules/.pnpm/@aws-lambda-powertools+logger@2.5.0/node_modules/@aws-lambda-powertools/logger/lib/cjs/formatter/PowertoolsLogFormatter.js:32:29)
    at Logger.createAndPopulateLogItem (/var/task/node_modules/.pnpm/@aws-lambda-powertools+logger@2.5.0/node_modules/@aws-lambda-powertools/logger/lib/cjs/Logger.js:722:39)
    at Logger.processLogItem (/var/task/node_modules/.pnpm/@aws-lambda-powertools+logger@2.5.0/node_modules/@aws-lambda-powertools/logger/lib/cjs/Logger.js:827:46)
    at Logger.info (/var/task/node_modules/.pnpm/@aws-lambda-powertools+logger@2.5.0/node_modules/@aws-lambda-power
```

**Comment by dreamorosi** [comment]
Hi @ralbertazzi - thank you for opening this issue.

I confirm that I am able to reproduce the issue: 
<img width="1158" alt="image" src="https://github.com/user-attachments/assets/64335276-a6fd-449b-a705-e9ce9e22fd30">

I agree also with the fix proposed, if the value is `:/etc/localtime` we should default to `:UTC`.  The fix should be applied in [this function](https://github.com/aws-powertools/powertools-lambda-typescript/blob/7a8848136b35e3bc6939374d7efc80252ebc3eec/packages/logger/src/formatter/LogFormatter.ts#L67-L79), which should be modified to do something like this:

```diff
public formatTimestamp(now: Date): string {
  const defaultTimezone = 'UTC';

  /**
   * If a specific timezone is configured and it's not the default `UTC`,
   * format the timestamp with the appropriate timezone offset.
   **/
- const configuredTimezone = this.envVarsService?.getTimezone();
+ const envVartimezone = this.envVarsService?.getTimezone();
+ const configuredTimezone =
+   envVartimezone === ':/etc/localtime' ? defaultTimezone : envVartimezone;
  if (configuredTimezone && !configuredTimezone.includes(defaultTimezone))
    return this.#generateISOTimestampWithOffset(now, configuredTimezone);

  return now.toISOString();
}
```

Additionally, we should also add a test case at the bottom of this block [here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/7a8848136b35e3bc6939374d7efc80252ebc3eec/packages/logger/tests/unit/formatter/PowertoolsLogFormatter.test.ts#L308) to test for this case, similar to:

```ts
it('uses the default UTC timezone when the TZ env variable is set to `:/etc/localtime`', () => {
  // Prepare
  process.env.TZ = ':/etc/localtime';
  const formatter = new PowertoolsLogFormatter({
    envVarsService: new EnvironmentVariablesService(),
  });

  // Act
  const timestamp = formatter.formatTimestamp(new Date());

  // Assess
  expect(timestamp).toEqual('2016-06-20T12:08:10.000Z');
});
```

I'll mark the issue as `help-wanted` and `good-first-issue` to signify that contributions are very welcome.

If nobody picks it up, we'll work on it and include the fix in the next release (~next Wednesday).

In the meantime, as a stopgap solution you can set the `TZ` environment variable to `:UTC` to continue using this version of Logger.

In parallel, I'll also consult the AWS Lambda team to see if the container image should align its behavior with the managed runtime & set the value to `:UTC`.

**Comment by daschaa** [comment]
I would like to take this issue :)
